graphql_compiler/macros/__init__.py

```diff
@@ -88,14 +97,23 @@ def register_macro_edge(macro_registry, macro_edge_graphql, macro_edge_args):
 def get_schema_with_macros(macro_registry):
     """Get a new GraphQLSchema with fields where macro edges can be used.
 
-    Specification:
+    Preconditions:
+    1. No macro in the registry has the same name as a field on the vertex where it applies,
+       or any interfaces it implements.
```

Comment:
This is redundant info. The fields of a vertex is a superset of the fields of the interface it implements.

Reply:
Agreed that it's redundant, but that's not necessarily a bad thing. We're going for making the code super clear and easy to understand, not for making the info fully normalized.